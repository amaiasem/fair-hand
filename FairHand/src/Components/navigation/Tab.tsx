import React from 'react'
import { StyleSheet, TouchableOpacity } from 'react-native'
import { AntDesign } from '@expo/vector-icons'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around'
  }
})

const Tab = ({ onPress, icon }: any) => {
  return (
    <TouchableOpacity style={styles.container} onPress={onPress} testID='button-tab'>
        {icon && <AntDesign name={icon} size={24} color="black" />}
    </TouchableOpacity>
  )
}

export default Tab
