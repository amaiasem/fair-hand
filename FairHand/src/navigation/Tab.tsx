import React from 'react'
import { StyleSheet, Text, TouchableOpacity } from 'react-native'
import { AntDesign } from '@expo/vector-icons'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around'
  }
})

const Tab = ({ tab, onPress, icon }) => {
  return (
        <TouchableOpacity style={styles.container} onPress={onPress}>
            <AntDesign name="home" size={24} color="black" />
            <Text>{tab.name}</Text>
        </TouchableOpacity>
  )
}

export default Tab
