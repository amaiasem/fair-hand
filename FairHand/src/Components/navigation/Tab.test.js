import React from 'react'
import { render } from '@testing-library/react-native'
import { NavigationContainer } from '@react-navigation/native'
import Tab from './Tab.tsx'

describe('Given a Tab component', () => {
  let component
  const onPress = jest.fn()
  const icon = 'home'
  beforeEach(() => {
    component = (
          <NavigationContainer>
              <Tab route={ { onPress, icon } } />
          </NavigationContainer>
    )
  })
  it('It should match the snapshot', () => {
    render(component)
    expect(component).toMatchSnapshot()
  })
})
