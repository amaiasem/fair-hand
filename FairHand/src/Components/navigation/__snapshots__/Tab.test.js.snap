// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Tab component It should match the snapshot 1`] = `
<ForwardRef(NavigationContainer)>
  <Tab
    route={
      Object {
        "icon": "home",
        "onPress": [MockFunction],
      }
    }
  />
</ForwardRef(NavigationContainer)>
`;
